app/controllers/hideout_controller.rb

```diff
@@ -0,0 +1,94 @@
+class HideoutController < ApplicationController
+  def create
+    # Needs owner_email and name as params.
+    # name must be lesser than 20 characters and greater than 3 characters
+    # owner must not already be in a hideout
+    # return the hideout as an object
+    owner_email = params[:owner_email]
+    name = params[:name]
+
+    if !User.exists?(email: owner_email)
+      render status: :not_found, body: "User does not exist"
+    elsif name.length > 20
+      render status: :precondition_failed, body: "Hideout name greater than 20 characters"
+    elsif name.length < 3
+      render status: :precondition_failed, body: "Hideout name lesser than 3 characters"
```

Comment:
Let's merge lines 12-13 into one single check for bad requests where the data is improper. Body of the response can simply be `Bad request` and error code should be `400`. Generally, that's the best practice I believe. Server responses shouldn't be too lengthy, we'll handle the specific length errors on the frontend.